upgrader.go

```diff
@@ -118,21 +119,25 @@ func newUpgrader(coordinationDir string, opts ...Option) (*Upgrader, error) {
 	return s, nil
 }
 
-func listenSock(coordinationDir string) (*net.UnixListener, error) {
-	listenpath := upgradeSockPath(coordinationDir, os.Getpid())
+func listenSock(osi osIface, coordinationDir string) (*net.UnixListener, error) {
+	listenpath := upgradeSockPath(coordinationDir, osi.Getpid())
 	return net.ListenUnix("unix", &net.UnixAddr{
 		Name: listenpath,
 		Net:  "unix",
 	})
 }
 
-func (u *Upgrader) AwaitUpgrade() error {
+var errClosed = errors.New("connection closed")
```

Comment:
I like to put these all at the top of the file they're in, makes them easier to find. If there are enough of them, I'd even put them in an `errors.go` file.

Reply:
For exported ones I definitely agree. For unexported ones, I think it's okay to put them closer to where they're used or at the top, either way.

Reply:
Fine with me.